opdracht-tech-it-easy/src/main/java/nl/novi/opdrachttechiteasy/service/TelevisionService.java

```diff
@@ -0,0 +1,71 @@
+package nl.novi.opdrachttechiteasy.service;
+
+
+import jakarta.persistence.EntityNotFoundException;
+import nl.novi.opdrachttechiteasy.exceptions.RecordNotFoundException;
+import nl.novi.opdrachttechiteasy.models.Television;
+import nl.novi.opdrachttechiteasy.repositories.TelevisionRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TelevisionService {
+
+
+    @Autowired
+    private final TelevisionRepository televisionRepository;
+
+    public TelevisionService(TelevisionRepository televisionRepository) {
+        this.televisionRepository = televisionRepository;
+    }
+
+
+
+    public List<Television> getTelevisions(){
+        return televisionRepository.findAll();
+    }
+
+    public Television getTelevision (Long id){
+        Optional<Television> optionalTelevision = televisionRepository.findById(id);
+        if (optionalTelevision.isEmpty()) {
+            throw new RecordNotFoundException();
+        } else {
+            return optionalTelevision.get();
+        }
+    }
+
+    public Television saveTelevision (Television television) {
+        return  televisionRepository.save(television);
+
+    }
+
+    public void removeTelevision(Long televisionId) {
+        System.out.println("the value of the id = " + televisionId);
+        if (!televisionRepository.existsById(televisionId)) {
+            throw new EntityNotFoundException("Television with ID " + televisionId + " not found");
+        }
+        televisionRepository.deleteById(televisionId);
+    }
+
+    public void updateTelevision (Long id, Television television) {
+
+        Optional<Television> televisionFound = televisionRepository.findById(id);
+        if (televisionFound.isPresent()) {
+            // not sure if there is a better way for doing this? Now I have to do this for each of them. Maybe a loop to make it easier?)
```

Comment:
Volgens mij heb ik in je vorige huiswerkopdracht al antwoord gegeven op deze vraag... 
Maar je kunt dit zeker extraheren in een herbruikbare methode. Dat zijn juist de leuke, nuttige uitdagingen die je aan wilt gaan om je code beter te maken. als eerste stap is dit echter voldoende, maar je moet daarna altijd nog refactoren.